Small fixes + improvements

hwoarangmy opened the pull request.

hwoarangmy committed:
Do not set what we can in the configure seat menu and refresh from server instead to avoid configuration desynchro

hwoarangmy committed:
Spawn portal rate should not be hardcoded
+ small improvements on portal animation handling
+ Do not let rogue portal spawn creatures

fixes #1119

hwoarangmy committed:
Use the same button for settings as the other buttons in the main menu

fixes #1109

hwoarangmy committed:
Integrated torture room object

part of #1117

hwoarangmy committed:
Improved flame particle effect due to node resizing

hwoarangmy committed:
Fixed workers being able to go through enemy doors

fixes #1112

Bertram25 commented on source/modes/MenuModeConfigureSeats.cpp:
```diff
@@ -160,12 +160,7 @@ void MenuModeConfigureSeats::activate()
                 CEGUI::ListboxTextItem* item = new CEGUI::ListboxTextItem(faction, cptFaction);
```
Commenting here as github removed per commit comment: SF! :)

akien-mga replied:
Oh that sucks :/

Bertram25 replied:
They need feed-back, so I'll try to contact them about this in a few days once I've got used to the new feature.
Did you guys had fun with the LAN? :)

hwoarangmy replied:
Yep. We played 2 games (me and @akien-mga). No segfault. We played "The bridge" but we found that gold is almost only in a shared part of the map. That means that you end up fighting to death with your 5 first creatures ^^

Bertram25 commented on config/rooms.cfg:
```diff
@@ -35,6 +35,8 @@
 # LibraryCooldownWorkMin            (uint)   How many turns a creature will stay idle after working minimum
 # LibraryCooldownWorkMax            (uint)   How many turns a creature will stay idle after working maximum
 # LibrarySkillPointsBook            (uint)   How many points have to be researched to create a book
+# PortalCooldownSpawnMin            (uint)   How many turns a portal will not spawn any creature after spawning minimum
```
As a user point of view, "... after spawning minimum" is not very straight-forward. I would simply say:
"How many minimum turns a portal will wait before spawning a creature"
and respectively:
"How many maximum turns a portal will wait before spawning a creature"

All that, if I understood the code correctly. ;)

hwoarangmy replied:
Yep, you understood correctly. I guess I used copy/paste too much and in this case, it is not very meaningfull ^^

Bertram25 commented on source/rooms/RoomPortal.cpp:
```diff
@@ -266,8 +270,6 @@ void RoomPortal::spawnCreature()
-    mSpawnCreatureCountdown = Random::Uint(30, 50);
 }
 
 void RoomPortal::setupRoom(const std::string& name, Seat* seat, const std::vector<Tile*>& tiles)
```
Looks good for the rest!

Bertram25 commented on gui/WindowSkirmishSubMenu.layout:
```diff
@@ -2,7 +2,7 @@
+        <Property name="Area" value="{{0.0,200},{0.5,-240},{0.0,400},{0.5,0}}" />
         <Property name="Text" value="" />
         <Property name="FrameEnabled" value="False" />
         <Property name="BackgroundEnabled" value="False" />
```
Code looks good

Bertram25 commented on source/rooms/RoomTorture.cpp:
```diff
@@ -129,7 +129,7 @@ BuildingObject* RoomTorture::notifyActiveSpotCreated(ActiveSpotPlace place, Tile
             Ogre::Real y = static_cast<Ogre::Real>(tile->getY());
             mCreaturesSpots.emplace(std::make_pair(tile, RoomTortureCreatureInfo()));
-            return loadBuildingObject(getGameMap(), "Roulette", tile, x, y, 0.0, SCALE_ACTIVESPOT_CENTER, false);
+            return loadBuildingObject(getGameMap(), "TortureObject", tile, x, y, 0.0, SCALE_ACTIVESPOT_CENTER, false);
```
Cool! :) No blocker left, bugfixes apart then?

hwoarangmy replied:
Well, @bark10 also reported to get his game frozen when he tries to open the settings menu. From what he said, I would guess that the modal window opens but is not visible for some reason. However, I have never seen that and I guess none of us did. If nobody did, we can still release and see if anyone else complains about this.

Bertram25 commented on particles/Flame.particle:
```diff
@@ -13,7 +13,7 @@ particle_system Flame
+        emission_rate   100
         time_to_live    1
         direction       0 1 0
         velocity    	0
```
Code looks good. I'll test that specifically.

Bertram25 commented on source/gamemap/GameMap.cpp:
```diff
@@ -1411,24 +1411,44 @@ bool GameMap::pathExists(const Creature* creature, Tile* tileStart, Tile* tileEn
+    }
 }
 
 std::list<Tile*> GameMap::path(int x1, int y1, int x2, int y2, const Creature* creature, Seat* seat, bool throughDiggableTiles)
```
Cool fix!

Bertram25 commented:
Mergeable as for me, my usual old-bear comments apart.

hwoarangmy committed:
Fixed not clear enough description of portal configurable spawn conditions

hwoarangmy committed:
Let building handle their building objects themselves instead of calling loadBuildingObject.

loadBuildingObject is not used by every building since persistent objects and traps. Now, it makes mandatory to change buildings when we change building objects. This dependency is not needed and pretty annoying so it is better to remove it

hwoarangmy committed:
Do not use a parameter telling if created entities are on server map but use the gamemap to know instead

For the record it was done like that to avoid the dependency between GameEntity and GameMap to allow unit testing of GameEntity. But since we removed this tests, there is no need to keep the boolean for every entity

hwoarangmy committed:
Save the trap in the door entity instead of using the trap name

hwoarangmy committed:
Door closing animation is not always correct at game load

hwoarangmy commented:
@Bertram25 I've fixed your usual old-bear comments + some code cleaning ;-)

Bertram25 commented on source/traps/TrapSpike.cpp:
```diff
@@ -146,5 +146,5 @@ bool TrapSpike::shoot(Tile* tile)
 TrapEntity* TrapSpike::getTrapEntity(Tile* tile)
 {
-    return new TrapEntity(getGameMap(), true, getName(), reg.getTrapFactory()->getMeshName(), tile, 0.0, true, isActivated(tile) ? 1.0f : 0.7f);
+    return new TrapEntity(getGameMap(), *this, reg.getTrapFactory()->getMeshName(), tile, 0.0, true, isActivated(tile) ? 1.0f : 0.7f);
```
Looks fine. Will have to test for regressions :)

Bertram25 commented on source/traps/TrapCannon.cpp:
```diff
@@ -148,7 +148,7 @@ bool TrapCannon::shoot(Tile* tile)
+    MissileOneHit* missile = new MissileOneHit(getGameMap(), getSeat(), getName(), "Cannonball",
         "", direction, ConfigManager::getSingleton().getTrapConfigDouble("CannonSpeed"),
         Random::Double(mMinDamage, mMaxDamage), 0.0, 0.0, nullptr, false, false);
     missile->addToGameMap();
```
Nice cleanup! Plus, I'd vote for some integration test instead if a similar unit test would enforce so much clutter.

hwoarangmy replied:
Yes, I guess some integration tests on traps would not be a waste

Bertram25 commented on source/entities/DoorEntity.cpp:
```diff
@@ -17,9 +17,10 @@
 
```
Less obvious, but better overall indeed.

hwoarangmy replied:
Before we had the game entity listener, I couldn't bind trap entities with traps because one might be deleted before the other one. For this reason, I had used a string to check if the trap was still alive before using it. Now, that we have the listener, we can use the trap directly without fear ^^

Bertram25 commented:
> @Bertram25 I've fixed your usual old-bear comments + some code cleaning ;-)

_Growl_ positively

Bertram25 commented:
I'll merge, the game is so nice now :) It feels complete in term of core feature.
